cinema/permissions.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+logger = logging.getLogger(__name__)
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    The request is authenticated as admin is a read/write (except delete) request, or
+    the request is authenticated a user is a read-only request.

```

Comment:
The docstring contains a grammatical error and is not very clear. It should be revised for clarity. Consider rephrasing it to: 'Allows read/write access for admin users (except delete), and read-only access for authenticated users.'